sycl/source/detail/platform_impl.hpp

```diff
@@ -191,12 +192,18 @@ class platform_impl {
   static std::shared_ptr<platform_impl>
   getPlatformFromPiDevice(RT::PiDevice PiDevice, const plugin &Plugin);
 
+  context getDefaultContext();
+
+  void pushDefaultContext(context Context);
+  void popDefaultContext();
+
 private:
   bool MHostPlatform = false;
   RT::PiPlatform MPlatform = 0;
   std::shared_ptr<plugin> MPlugin;
   std::vector<std::weak_ptr<device_impl>> MDeviceCache;
   std::mutex MDeviceMapMutex;
+  std::vector<context> MDefaultContextStack;
```

Comment:
That's a bit confusing. Variable name says it's a stack, but its usages indicate it's a vector. Why not use `std::stack`?

```suggestion
  std::stack<context> MDefaultContextStack;
```

Reply:
Probably just remove Stack from the name.  Choice of container isn't really important.

Reply:
Yes, that's also an options